doc/glossary.rst

```diff
@@ -96,4 +96,35 @@ Glossary of Terms
       term storage rather than copying it into RAM, saving writable memory for
       dynamic data and not the static program code.
 
+   Zephyr downstream module
+      A Zephyr module that the :file:`zephyr` repository default configuration
+      and default manifest file are **not** aware of.
+
+   Zephyr manifest module
+      A Zephyr module which also contains one or more west manifest files.
+
+   Zephyr module
+      A YAML file named :file:`zephyr/module.yml` and all the content it
+      references, which define the extensions to be hooked into the Zephyr
+      infrastructure.
+
+   Zephyr Project module
+      A Zephyr module managed by the Zephyr Project.  These modules will
+      be hosted at the Zephyr project organization on Github.
+
+   Zephyr upstream module
+      A Zephyr module that the :file:`zephyr` repository default configuration
+      and default manifest file are aware of.
+
+   :makevar:`ZEPHYR_MODULES`
+      The Zephyr build system CMake variable through which a list paths to
+      all Zephyr modules may be provided.
```

Comment:
what about:
```suggestion
     The Zephyr build system CMake variable through which a list of paths to
     Zephyr modules that should be included in the current build.
```

You could have `n`-Zephyr modules available, but decide to only include `m` of those in a particular build of projectA, while including `l` of them when building projectB.

In this case, the user probably will use a variant of 2. in this description: https://docs.zephyrproject.org/latest/guides/modules.html#without-west